core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -1996,6 +2059,104 @@ impl<R: WriteableRocks> StateManagerDatabase<R> {
+            };
+            let type_info = scrypto_decode::<TypeInfoSubstate>(new).expect("decode type info");
+
+            let entity_type = node_id.entity_type().expect("type of upserted ReNode");
+            let creation_id = CreationId::new(state_version, index_within_txn);
+
+            match type_info {
+                TypeInfoSubstate::Object(object_info) => {
+                    let blueprint_id = object_info.blueprint_info.blueprint_id;
+                    let BlueprintId {
+                        package_address,
+                        blueprint_name,
+                    } = blueprint_id.clone();
+                    db_context.cf(TypeAndCreationIndexedEntitiesCf).put(
+                        &(entity_type, creation_id.clone()),
+                        &EntityBlueprintId::of_object(*node_id, blueprint_id),
+                    );
+                    db_context.cf(BlueprintAndCreationIndexedObjectsCf).put(
+                        &(package_address, hash(&blueprint_name), creation_id),
+                        &ObjectBlueprintNameV1 {
+                            node_id: *node_id,
+                            blueprint_name,
+                        },
+                    );
+                }
+                TypeInfoSubstate::KeyValueStore(_kv_store_info) => {
+                    db_context.cf(TypeAndCreationIndexedEntitiesCf).put(
+                        &(entity_type, creation_id),
+                        &EntityBlueprintId::of_kv_store(*node_id),
+                    );
+                }
+                TypeInfoSubstate::GlobalAddressReservation(_)
+                | TypeInfoSubstate::GlobalAddressPhantom(_) => {
+                    panic!("should not be persisted: {:?}", type_info)
+                }
+            }
+        }
+    }
+
+    fn catchup_entity_listing_indices(&self) {
+        const TXN_FLUSH_INTERVAL: u64 = 10_000;
+
+        info!("ReNode listing indices are enabled.");
+        let db_context = self.open_rw_context();
+        let catchup_from_version = db_context
+            .cf(ExtensionsDataCf)
+            .get(&ExtensionsDataKey::EntityListingIndicesLastProcessedStateVersion)
+            .map(StateVersion::from_be_bytes)
+            .unwrap_or(StateVersion::pre_genesis())
+            .next()
+            .expect("next version");
+
+        let mut receipts_iter = db_context
+            .cf(TransactionReceiptsCf)
+            .iterate_from(&catchup_from_version, Direction::Forward)
+            .peekable();
+
+        while let Some((state_version, receipt)) = receipts_iter.next() {
+            self.batch_update_entity_listing_indices(
+                &db_context,
+                state_version,
+                &receipt.state_changes.substate_level_changes,
+            );
+            if state_version.number() % TXN_FLUSH_INTERVAL == 0 || receipts_iter.peek().is_none() {
+                info!(
+                    "ReNode listing indices updated to {}; flushing...",
```

Comment:
That's also a _lot!_ of info logs catching up on mainnet which has 75 million state versions right now.

Perhaps better to have info logs every 1 million transactions rather than every 10000? And then if you want logs every flush, perhaps make them `debug!` logs?